### PR TITLE
feat: Table and Column Lineage Polish

### PR DESCRIPTION
**Pull request by danwom**
### Summary of Changes
General UX polish for lineage features
- Loading state for column lineage
- Hide upstream/downstream tabs when not present

### Tests
- Added some tests about table detail tabs

### CheckList

Make sure you have checked **all** steps below to ensure a timely review.

- [ ] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [ ] PR includes a summary of changes, including screenshots of any UI changes.
- [ ] PR adds unit tests, updates existing unit tests, **OR** documents why no test additions or modifications are needed.
- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public python functions and the classes in the PR contain docstrings that explain what it does
- [ ] PR passes all tests documented in the [developer guide](https://github.com/lyft/amundsenfrontendlibrary/blob/master/docs/developer_guide.md#testing)

### CODE REVIEW

File: amundsen_application/static/js/pages/DashboardPage/ShimmeringDashboardLoader/index.tsx [hunk 1]
```diff
@@ -6,16 +6,16 @@ import * as React from 'react';
 import './styles.scss';
 
 const ShimmeringDashboardLoader: React.FC = () => (
```
Comment (danwom) [inline amundsen_application/static/js/pages/DashboardPage/ShimmeringDashboardLoader/index.tsx:8]: Using the pattern of "ShimmeringXLoader" seems somewhat verbose with both a prefix and a suffix. How do you feel about adopting either "XLoader" or "XShimmer" for these types of components?

**Comment by danwom** [comment]
@Golodhros @allisonsuarez 👀  😄

File: amundsen_application/static/js/pages/DashboardPage/ShimmeringDashboardLoader/index.tsx (hunk 1 above)
Comment (allisonsuarez) [reply amundsen_application/static/js/pages/DashboardPage/ShimmeringDashboardLoader/index.tsx:8]: What about just ShimmeringColumnLineageLoader?

**Review by allisonsuarez — APPROVED** [review]
LGTM

File: amundsen_application/static/js/features/ColumnList/ColumnLineageLoader/styles.scss [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+    .shimmer-loader-cell {
+      margin-bottom: $shimmer-block-spacing;
+
+      &:last-child {
+        margin-bottom: 0;
+      }
+
+      &.header {
+        height: 24px;
```
Comment (Golodhros) [inline amundsen_application/static/js/features/ColumnList/ColumnLineageLoader/styles.scss:36]:
```suggestion
        height: $shimmer-header-height;
```

File: amundsen_application/static/js/features/ColumnList/ColumnLineageLoader/styles.scss [hunk 3]
```diff
@@ -0,0 +1,46 @@
+// Copyright Contributors to the Amundsen project.
+// SPDX-License-Identifier: Apache-2.0
+
+@import 'variables';
+
+$shimmer-height: 120px;
+$shimmer-block-spacing: $spacer-2;
```
Comment (Golodhros) [inline amundsen_application/static/js/features/ColumnList/ColumnLineageLoader/styles.scss:7]: Nit: I feel this is a bit too much, I would make it spacer-1

File: amundsen_application/static/js/pages/DashboardPage/ShimmeringDashboardLoader/index.tsx (hunk 1 above)
Comment (Golodhros) [reply amundsen_application/static/js/pages/DashboardPage/ShimmeringDashboardLoader/index.tsx:8]: Sounds good!
Ideally, we should come up with a generic component that packs all this in one. At some point!

**Review by Golodhros — APPROVED** [review]
Some minor comments

**Pull request merged**